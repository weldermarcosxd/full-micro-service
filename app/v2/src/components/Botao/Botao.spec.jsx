import { render, screen } from "@testing-library/react";
import { Botao } from ".";
import userEvent from "@testing-library/user-event";

describe("<Botao />", () => {

    it("should render on screen with text", () => {
        render(<Botao text="Load More"></Botao>);
        expect.assertions(1);

        const botao = screen.getByRole("button", {name: /load more/i });
        expect(botao).toBeInTheDocument();
    });

    it("should call function on button click", () => {
        const fn = jest.fn();
        render(<Botao text="Load More" onClick={fn}></Botao>);
        expect.assertions(1);

        const botao = screen.getByRole("button", {name: /load more/i });
        userEvent.click(botao);

        expect(fn).toHaveBeenCalledTimes(1); 
    });

    it("should be disabled when disabled is true", () => {
        render(<Botao text="Load More" disabled={true}></Botao>);
        expect.assertions(1);

        const botao = screen.getByRole("button", {name: /load more/i });

        expect(botao).toBeDisabled();
    });

    it("should be enabled when disabled is false", () => {
        render(<Botao text="Load More"></Botao>);
        expect.assertions(1);

        const botao = screen.getByRole("button", {name: /load more/i });

        expect(botao).toBeEnabled();
    });

    it("should match snapshot", () => {
        const {container} = render(<Botao disabled={false} text="Load More"></Botao>);
        expect.assertions(1);
        expect(container.firstChild).toMatchSnapshot();
    });

});