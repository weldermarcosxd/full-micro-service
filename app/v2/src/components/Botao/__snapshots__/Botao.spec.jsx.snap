// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Botao /> should match snapshot 1`] = `
<button
  class="botao"
>
  Load More
</button>
`;
